### PR TITLE
Change z/TPF Machine Type Determination Logic

### PR DESCRIPTION
**Pull request by jjohnst-us**
Update existing z/TPF logic to determine Machine Type
Information to use the standard uname function 
instead of OS specific memory access.

Signed-off-by: jjohnst-us <jjohnst@us.ibm.com>

### CODE REVIEW

File: runtime/compiler/trj9/z/env/J9CPU.cpp [hunk 1]
```diff
@@ -209,11 +204,12 @@ CPU::TO_PORTLIB_get390zLinuxMachineType()
...
    const int pi1modArraySize = 5;
    char processorName[pi1modArraySize] = {0};
-   struct pi1dt *pid;
-   /* machine hardware name */
-   pid = (struct pi1dt *)cinfc_fast(CINFC_CMMPID);
-   sprintf(processorName, "%4.X", pid->pi1pids.pi1mslr.pi1mod);
-   int machine = atoi(processorName);
+
+   struct utsname info;
+   if (::uname(&info))
```
Comment (pshipton) [inline runtime/compiler/trj9/z/env/J9CPU.cpp]: Please add some curly brackets.

**Comment by jjohnst-us** [comment]
@pshipton Thanks Peter, update made.

**Comment by fjeremic** [comment]
I'm wondering if we can use the same implementation for the non-z/TPF platforms. We seem to use `FILE * fp = fopen("/proc/cpuinfo", "r");` right now which seems super dodgy.

**Comment by pshipton** [comment]
@jjohnst-us I don't see any update here

**Comment by pshipton** [comment]
Never mind, I see it now. The usually standard is to put the opening brace on the same line as the if statement.

**Comment by pshipton** [comment]
Ah, but this is JIT code, which I didn't realize so you can ignore that.

**Comment by jjohnst-us** [comment]
@pshipton Maybe I misunderstood the required update... I added line 210 & 212, is that what you meant by curly brace?

**Comment by jjohnst-us** [comment]
yah I followed suit with other blocks in the same segment.

**Comment by jjohnst-us** [comment]
@fjeremic could you review this PR, I think @pshipton said he was deferring to someone on JIT team.  Thanks guys.

File: runtime/compiler/trj9/z/env/J9CPU.cpp [hunk 2]
```diff
@@ -209,11 +204,14 @@ CPU::TO_PORTLIB_get390zLinuxMachineType()
...
+      {
+	  return ret_machine;
```
Comment (fjeremic) [inline runtime/compiler/trj9/z/env/J9CPU.cpp]: There seems to be a tab here. Also can you fix the indentation to:
```
   if (::uname(&info))
      {
      return ret_machine;
      }
```

File: runtime/compiler/trj9/z/env/J9CPU.cpp [hunk 3]
```diff
@@ -209,11 +204,14 @@ CPU::TO_PORTLIB_get390zLinuxMachineType()
...
-   sprintf(processorName, "%4.X", pid->pi1pids.pi1mslr.pi1mod);
-   int machine = atoi(processorName);
+
+   struct utsname info;
+   if (::uname(&info))
+      {
+      return ret_machine;
+      }
+
+   uint32_t machine = atoi(info.machine);
```
Comment (joransiu) [inline runtime/compiler/trj9/z/env/J9CPU.cpp]: Just to confirm, does `utsname.machine` return the 4-digit machine code?  When I tried it on RedHat/SLES on Z, I got `s390x` instead.

Comment (jjohnst-us) [reply runtime/compiler/trj9/z/env/J9CPU.cpp]: @joransiu 
I created a test program on our platform and here is the output:
CSMP0097I 11.00.55 CPU-B SS-BSS  SSU-HPN  IS-01   
FILE0001I 11.00.55 START OF DISPLAY FROM myprogram
uname machine field response was: 3906-7D1        
END OF DISPLAY+ 
Here is the program:
   struct utsname info;
   if (uname(&info))
      {
      printf("uname call failed!");
      return -1;
      }

   printf("uname machine field response was: %s", info.machine);

Comment (jjohnst-us) [reply runtime/compiler/trj9/z/env/J9CPU.cpp]: Hmm after looking at that code again I realized there are two lines from the old way we were doing it that aren't necessary anymore:
   const int pi1modArraySize = 5;
   char processorName[pi1modArraySize] = {0};

I'll remove those and re-push.

Comment (joransiu) [reply runtime/compiler/trj9/z/env/J9CPU.cpp]: Thanks @jjohnst-us for confirming!

**Comment by jjohnst-us** [comment]
@joransiu @fjeremic 
Sounds like everyone is okay with this update?  Anything else required to get this PR merged?
tia

**Comment by fjeremic** [comment]
Jenkins test sanity zlinux

**Comment by jjohnst-us** [comment]
@pshipton, looks like we are all set for merge, tia.

**Comment by pshipton** [comment]
Someone on the jit team who has reviewed the changes should do the merge.

**Pull request merged**